pykafka/simpleconsumer.py

```diff
@@ -206,6 +206,7 @@ def _update(self):
         with self._update_lock:
             self._cluster.update()
             self._setup_partitions_by_leader()
+            self._discover_offset_manager()
```

Comment:
Removing [this line](https://github.com/Parsely/pykafka/blob/2b3a9f14880f20649b076da156c00a4aefc2e55a/pykafka/cluster.py#L242) does solve the issue, though I don't think that's the best solution. Ideally we'd figure out what's holding a reference to the old `Broker` instance, but I'm having a really hard time doing that.